### PR TITLE
Change maven site report name and description

### PR DESCRIPTION
**Pull request by chonton**
Change the maven site japicmp report to be more descriptive.  The report name is more consistent with other reports like PMD, Findbugs, and Checkstyle.  The report description includes the old and new versions.

### CODE REVIEW

File: japicmp-maven-plugin/src/main/java/japicmp/maven/JApiCmpMojo.java [hunk 1]
```diff
@@ -89,6 +89,7 @@
 	private String versionRangeWithProjectVersion;
 	@Component
 	private ArtifactMetadataSource metadataSource;
+	private Options options;
```
Comment (chonton) [inline japicmp-maven-plugin/src/main/java/japicmp/maven/JApiCmpMojo.java]: refactorings to allow access to final options from the report mojo

File: japicmp-maven-plugin/src/main/java/japicmp/maven/JApiCmpReport.java [hunk 2]
```diff
@@ -58,13 +61,14 @@
...
+	private MavenParameters mavenParameters;
+	private PluginParameters pluginParameters;
 
 	@Override
 	protected void executeReport(Locale locale) throws MavenReportException {
 		try {
-			JApiCmpMojo mojo = new JApiCmpMojo();
-			MavenParameters mavenParameters = new MavenParameters(artifactRepositories, artifactFactory, localRepository, artifactResolver, mavenProject, mojoExecution, versionRangeWithProjectVersion, metadataSource);
-			PluginParameters pluginParameters = new PluginParameters(skip, newVersion, oldVersion, parameter, dependencies, Optional.<File>absent(), Optional.of(outputDirectory), false, oldVersions, newVersions, oldClassPathDependencies, newClassPathDependencies);
+			JApiCmpMojo mojo = getMojo();
```
Comment (chonton) [inline japicmp-maven-plugin/src/main/java/japicmp/maven/JApiCmpReport.java]: refactored to 'cache' options

File: japicmp-maven-plugin/src/main/java/japicmp/maven/JApiCmpReport.java [hunk 3]
```diff
@@ -90,18 +94,54 @@ protected void executeReport(Locale locale) throws MavenReportException {
...
+			return getMojo().getOptions(pluginParameters, mavenParameters);
+		} catch (MojoFailureException ignore) {
+			return null;
+		}
+	}
+
 	@Override
 	public String getOutputName() {
-		return "japicmp-maven-plugin-report";
+		return "japicmp";
```
Comment (chonton) [inline japicmp-maven-plugin/src/main/java/japicmp/maven/JApiCmpReport.java]: more consistent with other reports such as PMD, Checkstyle, Findbugs...

File: japicmp-maven-plugin/src/main/java/japicmp/maven/JApiCmpReport.java [hunk 4]
```diff
@@ -90,18 +94,54 @@ protected void executeReport(Locale locale) throws MavenReportException {
...
 	}
 
 	@Override
 	public String getName(Locale locale) {
-		return "japicmp-maven-plugin";
+		return "japicmp";
```
Comment (chonton) [inline japicmp-maven-plugin/src/main/java/japicmp/maven/JApiCmpReport.java]: more consistent with other reports such as PMD, Checkstyle, Findbugs...

File: japicmp-maven-plugin/src/main/java/japicmp/maven/JApiCmpReport.java [hunk 5]
```diff
@@ -90,18 +94,54 @@ protected void executeReport(Locale locale) throws MavenReportException {
...
 	}
 
 	@Override
 	public String getDescription(Locale locale) {
-		return "japicmp is a maven plugin that computes the differences between two versions of a jar file/artifact.";
```
Comment (chonton) [inline japicmp-maven-plugin/src/main/java/japicmp/maven/JApiCmpReport.java]: Description includes the old and new versions.

File: japicmp-testbase/japicmp-test-maven-plugin/src/test/java/japicmp/test/ITReportTitle.java [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+		assertThat(leftNav.attr("title"), is("japicmp"));
+		assertThat(leftNav.text(), is("japicmp"));
+
+		Elements overviewRow = document.select("#bodyColumn tr:has([href=\"japicmp.html\"])");
+		Elements link = overviewRow.select("[href=\"japicmp.html\"]");
+		assertThat(link.text(), is("japicmp"));
+
+		Elements description = overviewRow.select("td:eq(1)");
+		assertThat(description.text(), is("Source compatibility of japicmp-test-v2-0.8.2-SNAPSHOT.jar against japicmp-test-v2-0.8.2-SNAPSHOT.jar"));
+	}
```
Comment (chonton) [inline japicmp-testbase/japicmp-test-maven-plugin/src/test/java/japicmp/test/ITReportTitle.java]: test maven project-reports has the correct name and description of japicmp report.

Comment (siom79) [inline japicmp-testbase/japicmp-test-maven-plugin/src/test/java/japicmp/test/ITReportTitle.java]: The version of the two jar archives are hard-coded.

**Pull request merged**